packages/fx-core/src/common/tools.ts

```diff
@@ -110,23 +150,28 @@ export function sperateSecretData(configJson: Json): Record<string, string> {
     if (!resourceConfig) continue;
     if ("*" !== item) {
       const configValue = resourceConfig[item];
-      if (configValue) {
+      if (
+        configValue &&
+        !((configValue as string).startsWith("{{") && (configValue as string).endsWith("}}"))
+      ) {
         const keyName = `${resourceId}.${item}`;
-        res[keyName] = configValue;
+        dict[keyName] = configValue;
         resourceConfig[item] = `{{${keyName}}}`;
       }
     } else {
       for (const itemName of Object.keys(resourceConfig)) {
         const configValue = resourceConfig[itemName];
-        if (configValue !== undefined) {
+        if (
+          configValue !== undefined &&
```

Comment:
We can add a helper function for this if condition because it appear multiple times